exporter/metric/metric_test.go

```diff
@@ -118,3 +118,87 @@ func TestRecordToMpb(t *testing.T) {
 		t.Errorf("%v", aggError)
 	}
 }
+
+
+func TestResourceToMonitoredResourcepb(t *testing.T) {
+	resList := []*resource.Resource{
+	// correct k8s resources
+	resource.New(
+		kv.String("type", "k8s"),
+		kv.String("cloud.zone", "us-central1-a"),
+		kv.String("k8s.cluster.name", "opentelemetry-cluster"),
+		kv.String("k8s.namespace.name", "default"),
+		kv.String("k8s.pod.name", "opentelemetry-pod-autoconf"),
+		kv.String("k8s.deployment.name", "opentelemetry"),		
+	), 
+	// nonexisting resource types
+	resource.New(
+		kv.String("type", "none"),
+		kv.String("cloud.zone", "us-central1-a"),
+		kv.String("k8s.cluster.name", "opentelemetry-cluster"),
+		kv.String("k8s.namespace.name", "default"),
+		kv.String("k8s.pod.name", "opentelemetry-pod-autoconf"),
+		kv.String("k8s.deployment.name", "opentelemetry"),		
+	),
+	// missing resource fields
+	resource.New(
+		kv.String("type", "k8s"),
+		kv.String("cloud.zone", "us-central1-a"),
+		kv.String("k8s.cluster.name", "opentelemetry-cluster"),
+		kv.String("k8s.namespace.name", "default"),
+		kv.String("k8s.pod.name", "opentelemetry-pod-autoconf"),
+	)}
+
+	expectedTypes := []string{"k8s_container","global", "global"}
+
+	expectedLabels := []map[string]string {
+		map[string]string{
+			"project_id":     "",
+			"location": "us-central1-a",
+			"cluster_name": "opentelemetry-cluster",
+			"namespace_name": "default",
+			"pod_name": "opentelemetry-pod-autoconf",
+			"container_name": "opentelemetry",
+		},
+		map[string]string{
+			"project_id":     "",
+		},
+		map[string]string{
```

Comment:
Should we always ignore the entire k8s type if a field is missing? Maybe parsing partial data is fine. What do you think?

Reply:
I see. In fact, many k8s Monitored resources like k8s_node, k8s_pod, and k8s_cluster, share some common fields. (https://cloud.google.com/monitoring/api/resources#tag_k8s_cluster) I think we need to follow the definition strictly otherwise we may misjudge a type of Monitored resources.

Reply:
Historically in Opencensus we only use k8s_container (https://cloud.google.com/monitoring/api/resources#tag_k8s_container) resource type as it already consists of all the related fields for Kubernetes. It would take additional logic to parse by cluster, nodes, pods, etc. By having one resource type for k8s, parsing the resource labels would be much easier in the Cloud Trace backend as well.

Reply:
I see. I would allow missing fields for more flexible types of monitored resources